From 5176cb4c8bdc6ee2585f82901acb4a04ee4e65fa Mon Sep 17 00:00:00 2001
From: Marco Martin <notmart@gmail.com>
Date: Thu, 24 Mar 2022 15:16:15 +0100
Subject: [PATCH] Backport fixing the writing of lnf defaults

properly write the config files under kdedefaults/ when switching
the look and fell package
---
 kcms/lookandfeel/kcm.cpp | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/kcms/lookandfeel/kcm.cpp b/kcms/lookandfeel/kcm.cpp
index a80719a05..cb789e1e9 100644
--- a/kcms/lookandfeel/kcm.cpp
+++ b/kcms/lookandfeel/kcm.cpp
@@ -831,11 +831,7 @@ void KCMLookandFeel::writeNewDefaults(KConfig &config,
 
 void KCMLookandFeel::writeNewDefaults(KConfigGroup &cg, KConfigGroup &cgd, const QString &key, const QString &value, KConfig::WriteConfigFlags writeFlags)
 {
-    if (m_data->isDefaults()) {
-        cgd.revertToDefault(key);
-    } else {
-        cgd.writeEntry(key, value, writeFlags);
-    }
+    cgd.writeEntry(key, value, writeFlags);
     cgd.sync();
 
     cg.revertToDefault(key, writeFlags);
-- 
2.35.0

